### PR TITLE
default_coafile: Change default_coafile naming convention to system_coafile

### LINKED ISSUE #3389: Change default_coafile naming convention.
Should we change the name of `default_coafile` in `coalib` to `system_coafile`?
It will avoid confusion due to the following lines in `Constants.py`
```
system_coafile = os.path.join(coalib_root, 'default_coafile')

user_coafile = os.path.join(os.path.expanduser('~'), '.coarc')

default_coafile = '.coafile'
```

### PR DESCRIPTION
**Pull request by pradeepgangwar**
Changes naming convention of deafult_coafile and now uses system_coafile as the configuration file.

Closes #3389

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

### CODE REVIEW

File: coalib/misc/Constants.py [hunk 1]
```diff
@@ -109,7 +110,11 @@
                                     'languages',
                                     'definitions')
 
-system_coafile = os.path.join(coalib_root, 'default_coafile')
+if os.path.isfile(os.path.join(coalib_root, 'system_coafile')):
+    system_coafile = os.path.join(coalib_root, 'system_coafile')
+else:
+    system_coafile = os.path.join(coalib_root, 'default_coafile')
+    logging.warning('deprecated')
```
Comment (Mixih) [inline coalib/misc/Constants.py]: need to use logging.getLogger before this and call the warning on the logger object it returns. This makes sure that the config is applied correctly. Also, the deprecation message needs to be more descriptive

File: tests/settings/ConfigurationGatheringTest.py [hunk 2]
```diff
@@ -83,13 +83,13 @@ def test_gather_configuration(self):
 
         self.assertEqual(len(local_bears['cli']), 0)
 
-    def test_default_coafile_parsing(self):
+    def test_system_coafile_parsing(self):
```
Comment (Mixih) [inline tests/settings/ConfigurationGatheringTest.py]: you need to test that the old file name still works and outputs a deprecated message.

**Comment by Mixih** [comment]
also there should be a newline between the "closes" and the last commit body line.

**Comment by pradeepgangwar** [comment]
@Mixih @Makman2 I don't know why tests fails on CircleCI and appveyor saying 
```
Expected:
    WARNING:root:The bear SomeOldBear is deprecated. Use SomeBear instead!
    I'm running!
Got:
    I'm running!
```
Please let me know it this PR requires more changes.

**Comment by pradeepgangwar** [comment]
Please review changes. Tests are successful.

File: tests/settings/ConfigurationGatheringTest.py [hunk 3]
```diff
@@ -83,13 +84,39 @@ def test_gather_configuration(self):
 
         self.assertEqual(len(local_bears['cli']), 0)
 
-    def test_default_coafile_parsing(self):
+    def test_default_coafile_deprecation(self):
+        '''
+        This test is temporary, it will send a directory
+        path which doesnot contain system_coafile named file
```
Comment (Makman2) [inline tests/settings/ConfigurationGatheringTest.py]: `doesn't` or `does not` ;)

File: tests/settings/ConfigurationGatheringTest.py [hunk 4]
```diff
@@ -83,13 +84,39 @@ def test_gather_configuration(self):
...
+        to function in Constants.py and ask for allocating
+        a system_coafile. Since there is no file nameds
```
Comment (Makman2) [inline tests/settings/ConfigurationGatheringTest.py]: named~~s~~

File: tests/settings/ConfigurationGatheringTest.py [hunk 5]
```diff
@@ -83,13 +84,39 @@ def test_gather_configuration(self):
...
+        system_coafile, it should log a warning
+        '''
```
Comment (Makman2) [inline tests/settings/ConfigurationGatheringTest.py]: Optional:
I think the test is quite self-explaining mentioning some deprecation of `default_coafile`, so you don't have to put it here :)

File: tests/settings/ConfigurationGatheringTest.py [hunk 6]
```diff
@@ -83,13 +84,39 @@ def test_gather_configuration(self):
...
+        tmp = Constants.system_coafile
+        logger = logging.getLogger()
+
+        system_coafile_path = os.path.abspath(os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            'section_manager_test_files',
+            'child_dir'))
+
+        with self.assertLogs(logger, 'WARNING') as cm:
+            get_system_coafile(system_coafile_path)
```
Comment (Makman2) [inline tests/settings/ConfigurationGatheringTest.py]: you can use `LogCapture` if you like, then you don't have to retrieve the logger with `getLogger()` ;)

File: tests/settings/ConfigurationGatheringTest.py [hunk 7]
```diff
@@ -83,13 +84,39 @@ def test_gather_configuration(self):
...
+    def test_default_coafile_deprecation(self):
+        '''
+        This test is temporary, it will send a directory
+        path which doesnot contain system_coafile named file
+        to function in Constants.py and ask for allocating
+        a system_coafile. Since there is no file nameds
+        system_coafile, it should log a warning
+        '''
+
+        tmp = Constants.system_coafile
```
Comment (Makman2) [inline tests/settings/ConfigurationGatheringTest.py]: unused variable

File: tests/settings/ConfigurationGatheringTest.py [hunk 8]
```diff
@@ -83,13 +84,39 @@ def test_gather_configuration(self):
...
+            'child_dir'))
+
+        with self.assertLogs(logger, 'WARNING') as cm:
+            get_system_coafile(system_coafile_path)
+
+        self.assertEqual(cm.output, [
+            "WARNING:root:Filename 'deafult_coafile' has been deprecated. "
+            "Please use 'system_coafile' instead."])
+
+        Constants.system_coafile = tmp
```
Comment (Makman2) [inline tests/settings/ConfigurationGatheringTest.py]: obsolete assignment

File: tests/settings/ConfigurationGatheringTest.py (hunk 6 above)
Comment (Makman2) [reply tests/settings/ConfigurationGatheringTest.py]: together with `LogCapture.check` which works imo a bit simpler than asserting for `cm.output` ;)

**Comment by Makman2** [comment]
Looks good basically, now we have to defeat only minor issues ;)

**Comment by pradeepgangwar** [comment]
@Makman2 I have fixed code based on comments. Please review.

**Comment by Makman2** [comment]
ack de53fc9

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
@gitmate-bot ff

**Pull request merged**